cmd/merge-sql-schemas/main.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"database/sql"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sort"
+
+	_ "modernc.org/sqlite" // pure Go SQLite driver
+)
+
+func main() {
+	// Open an in-memory SQLite database.
+	db, err := sql.Open("sqlite", ":memory:")
+	if err != nil {
+		log.Fatalf("failed to open in-memory db: %v", err)
+	}
+	defer db.Close()
+
+	migrationDir := "tapdb/sqlc/migrations"
+	files, err := ioutil.ReadDir(migrationDir)
+	if err != nil {
+		log.Fatalf("failed to read migration dir: %v", err)
+	}
+
+	var upFiles []string
+	upRegex := regexp.MustCompile(`\.up\.sql$`)
+	for _, f := range files {
+		if !f.IsDir() && upRegex.MatchString(f.Name()) {
+			upFiles = append(upFiles, f.Name())
+		}
+	}
+	sort.Strings(upFiles)
+
+	// Execute each up migration in order.
+	for _, fname := range upFiles {
+		path := filepath.Join(migrationDir, fname)
+		data, err := ioutil.ReadFile(path)
+		if err != nil {
+			log.Fatalf("failed to read file %s: %v", fname, err)
+		}
+		_, err = db.Exec(string(data))
+		if err != nil {
+			log.Fatalf("error executing migration %s: %v", fname,
+				err)
+		}
+	}
+
+	// ---------------------------------------------------------------------
+	// Retrieve final database schema from sqlite_master.
+	//
+	// SQLite automatically maintains a special table called sqlite_master,
+	// which holds metadata about all objects inside the database, such as
+	// tables, views, indexes, and triggers. Each row in this table
+	// represents an object, with columns such as "type" (the kind of
+	// object), "name" (the object's name), and "sql" (the SQL DDL statement
+	// that created it).
+	//
+	// In our case, after running all the migration files on an in‑memory
+	// database, we execute the following query to extract only the schema
+	// definitions for tables and views. Ordering by name ensures the output
+	// is stable across runs.
+	//
+	// This way, we can consolidate and export the complete database schema
+	// as it stands after all migrations have been applied.
+	// ---------------------------------------------------------------------
+	rows, err := db.Query(`SELECT type, name, sql FROM sqlite_master WHERE type IN ('table','view') ORDER BY name`) //nolint:lll
```

Comment:
nit: why not just break the string into two lines instead of making things even longer with the `nolint` comment?

Reply:
Ended up making it multi-line given it's using a multi-line string literal.